### PR TITLE
monitoring: add PanelOptions.ZeroIfNoData

### PR DESCRIPTION
**Pull request by bobheadxi**
ZeroIfNoData adjusts an observable's query such that "no data" will render as "0". This is useful if your observable tracks error rates, which might show "no data" if all is well and there are no errors.

I only applied this to a single observable right now, to close https://github.com/sourcegraph/sourcegraph/issues/20041

### CODE REVIEW

File: monitoring/monitoring/panel_options.go [hunk 1]
```diff
@@ -224,3 +224,19 @@ func (panelOptionsLibrary) NoLegend() ObservablePanelOption {
...
+// This is useful if your observable tracks error rates, which might show "no data" if
+// all is well and there are no errors.
+//
+// This is different from Grafana's "null as zero", since "no data" is not "null".
+func (panelOptionsLibrary) ZeroIfNoData() ObservablePanelOption {
+	return func(o Observable, p *sdk.Panel) {
+		switch o.Panel.panelType {
+		case PanelTypeGraph:
+			p.GraphPanel.Targets[0].Expr += " OR on() vector(0)"
+		case Panel().panelType:
```
Comment (ryanslade) [inline monitoring/monitoring/panel_options.go:238]: Is this correct? Shouldn't t be something like:

```go
case PanelTypeHeatmap:
```

File: monitoring/monitoring/generator.go [hunk 2]
```diff
@@ -74,8 +74,12 @@ func Generate(logger log15.Logger, opts GenerateOptions, containers ...*Containe
 			if opts.Reload {
 				crlog := clog.New("instance", localGrafanaURL)
 				crlog.Debug("Reloading Grafana instance")
-				client := sdk.NewClient(localGrafanaURL, localGrafanaCredentials, sdk.DefaultHTTPClient)
-				_, err := client.SetDashboard(context.Background(), *board, sdk.SetDashboardParams{Overwrite: true})
+				client, err := sdk.NewClient(localGrafanaURL, localGrafanaCredentials, sdk.DefaultHTTPClient)
```
Comment (ryanslade) [inline monitoring/monitoring/generator.go:77]: Why do we need this change?

**Review by ryanslade — APPROVED** [review]
Happy with this in principal, just unsure about a couple of specific changes

File: monitoring/monitoring/panel_options.go (hunk 1 above)
Comment (bobheadxi) [reply monitoring/monitoring/panel_options.go:238]: omg

Comment (bobheadxi) [reply monitoring/monitoring/panel_options.go]: typed too fast for the autocomplete, thanks for the catch! e212d20

File: monitoring/monitoring/generator.go (hunk 2 above)
Comment (bobheadxi) [reply monitoring/monitoring/generator.go:77]: it seems the API for the grafana sdk's client library changed

**Pull request merged**